### PR TITLE
Bug 1215546 - Fixes add-on list after installation, r=gasolin

### PR DESCRIPTION
**Pull request by scottwu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7571941e3585ef3b8e2bafa209dc642e357dbeb2

File: apps/settings/js/modules/addon_manager.js [hunk 1]
```diff
@@ -43,7 +43,11 @@ define(function(require) {
      * @memberOf AddonManager.prototype
      */
     _bindEvents: function() {
-      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
+      AppsCache.addEventListener('oninstall', (evt) => {
+        evt.application.addEventListener('downloadsuccess', (evt) => {
+          this._updateAddons(evt);
+        });
+      });
       AppsCache.addEventListener('onuninstall', this._updateAddons.bind(this));
```
Comment (gasolin) [inline apps/settings/js/modules/addon_manager.js]: mmm, seems current code is not ideal, we could create a _boundUpdateAddons object to bind _updateAddons and reuse the same object to handle these 3 events

File: apps/settings/js/modules/addon_manager.js [hunk 2]
```diff
@@ -43,7 +43,11 @@ define(function(require) {
      * @memberOf AddonManager.prototype
      */
     _bindEvents: function() {
-      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
+      AppsCache.addEventListener('oninstall', (evt) => {
+        evt.application.addEventListener('downloadsuccess', (evt) => {
```
Comment (gasolin) [inline apps/settings/js/modules/addon_manager.js]: please add some comment according to what fabrice provided, so we won't forget it after several months

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9a09e0518af2219d367094b40ffd0e9039c2c9c2

File: apps/settings/js/modules/addon_manager.js [hunk 3]
```diff
@@ -43,10 +43,17 @@ define(function(require) {
      * @memberOf AddonManager.prototype
      */
     _bindEvents: function() {
-      AppsCache.addEventListener('oninstall', this._updateAddons.bind(this));
-      AppsCache.addEventListener('onuninstall', this._updateAddons.bind(this));
-      mozApps.mgmt.addEventListener('enabledstatechange',
-        this._updateAddons.bind(this));
+      var _updateAddons = this._updateAddons.bind(this);
```
Comment (gasolin) [inline apps/settings/js/modules/addon_manager.js]: let's use another variable `_boundUpdateAddons` instead

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a0d2f7039a2673a6340a9e2e426a23a8d64e6456

**Pull request merged**